godtools/App/Share/Domain/UseCases/GetAllToolsUseCase/GetAllToolsUseCase.swift

```diff
@@ -17,6 +17,24 @@ class GetAllToolsUseCase {
         self.getToolUseCase = getToolUseCase
         self.resourcesRepository = resourcesRepository
     }
+}
+
+// MARK: - Publishers
+
+extension GetAllToolsUseCase {
+    
+    func getToolsForCategoryPublisher(category: CurrentValueSubject<String?, Never>) -> AnyPublisher<[ToolDomainModel], Never> {
+        
+        return Publishers.CombineLatest(resourcesRepository.getResourcesChanged(), category)
+            .flatMap { (_, category) -> AnyPublisher<[ToolDomainModel], Never> in
+                
+                let tools = self.getAllTools(sorted: true, with: category)
+                
+                return Just(tools)
+                    .eraseToAnyPublisher()
+            }
+            .eraseToAnyPublisher()
+    }
     
     func getAllToolsPublisher(sorted: Bool) -> AnyPublisher<[ToolDomainModel], Never> {
```

Comment:
Hey @rachaelblue I think this method should be private since use cases should expose a single point of execution.  The GetAllToolsUseCase we will always want to filter on the category since that is the intended functionality.  And I like the idea of passing the category CurrentValueSubject and using CombineLatest here.

Reply:
That method wasn't being used anymore so I deleted it.  But, your comment about having only one method exposed per use case brings me to a different question -- `GetAllToolsUseCase` has an additional public method that returns an array of `ResourceModel`s because `GetToolCategoriesUseCase` needs the all tools result before it gets mapped to `ToolDomainModel`.  Thoughts about that?  This may be one potential reason for putting certain filtering logic on the repository itself, when multiple use cases need to use it.

Reply:
Also, in order to move the logic that adds the "All Tools" category into the use case, I made a dummy id to keep track of that category- `static let allToolsCategoryId = "allTools"`.  I have this in the `GetToolCategoriesUseCase`, but I'm not sure where else it might fit better.  Does that make sense?